src/extension/serializer.ts

```diff
@@ -596,9 +601,18 @@ export class GrpcSerializer extends SerializerBase {
   protected applyIdentity(data: Notebook): Notebook {
     const identity = this.lifecycleIdentity
     switch (identity) {
-      case RunmeIdentity.UNSPECIFIED:
       case RunmeIdentity.DOCUMENT:
         break
+      case RunmeIdentity.UNSPECIFIED:
+        data.cells.forEach((cell) => {
+          if (cell.kind !== CellKind.CODE) {
+            return
+          }
+          if (cell.metadata?.['id']) {
+            cell.metadata['runme.dev/originalId'] = cell.metadata['id']
```

Comment:
Don't think this is the right solution. It tries to fix "the issue" at the tail end of serialization instead of deserialization where it seems to originate.

I believe this code here's not "smart enough" yet. It ignores both identity requirements and the fact that `runme.dev/id` is not set which means cell identity wasn't required in the first place.
https://github.com/stateful/vscode-runme/blob/lifecycle-id-fixes/src/extension/serializer.ts#L319-L324

Reply:
One could argue that if `runme.dev/id` is not set the deserialization was requested without cell identity.